### PR TITLE
Adds a new status effect that halts slowdown when fighting megafauna

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
balance: When in combat with megafauna, you now gain the Adrenaline status effect, which prevents slowdowns.
/:cl:

Against nearly all megafauna, you have about 50 or so effective health. Once you drop below that point, your movement is slowed - which makes you unable to dodge most attacks and thus leads to a quick and painful death without a stockpile of healing cores or a survival medipen handy.

While a megafauna is attacking a target, all cliented mobs that can be targeted by it now gain the Adrenaline status effect. This effect applies `IGNORESLOWDOWN`, meaning that they will move at their normal speed regardless of health level. The status effect lasts for three seconds (refreshed every `Life()` tick of that megafauna), so if you flee from them the slowdown from your wounds will quickly set in.

This is mainly so that lagging out for half a second doesn't make it completely impossible to win against a megafauna, as well as allowing for more tense gameplay moments where you're very close to death but the creature is on its last legs and fighting for its life.

As Ergo is generally referred to for all mining-related changes, I will now say his name three times into a mirror to summon him from the infinite beyond.

@Ergovisavi 
@Ergovisavi 
@Ergovisavi

### CODE REVIEW

**Comment by Repukan** [comment]
Fuck no.

The whole point to fighting them is figuring out the attack pattern. This is like adding a accelerated thermal evolution everytime someone is near lit plasmafires.

**Comment by Repukan** [comment]
And if you're lagging and you die to megafauna, this is not the fault of the game design. It is your fault for knowingly fighting the big bad knowing you might stutter or that your toaster can't handle it

**Comment by Xhuis** [comment]
Attack patterns are circumvented by the way the mob's AI plays out. With the life tick system it can take a chunk out of you just from being next to it, in addition to high time dilation and lagspikes making it nigh-impossible to avoid attacks or certain telegraphs inaccurate and misleading. Punishing players for unavoidable damage or because Comcast is having a bad day is something that should be avoided.

**Comment by Repukan** [comment]
You can't code against lag or natural occurrences.

If you're lagging and don't want to die, then don't fight it. You just make it easy to abuse (both the megafauna and the loot it protects)

**Comment by Repukan** [comment]
To say it's unavoidable it's false because they have 1) natural lighting 2) a GPS signal 3) telegraphed attacks.

**Comment by Xhuis** [comment]
I still disagree with both of your points, for reasons I've outlined. I'll see what Ergo thinks; he's more experienced in this than either of us.

**Comment by KorPhaeron** [comment]
Feels weird and gamey and kind of impossible to die like this

Can't they get stims from their pen or from chemistry if they want to ignore slowdowns?

**Comment by Xhuis** [comment]
It could work; this PR sparked from a discussion on coderbus about two months back that I still felt relevant. If ergo feels the same way, I'll close this.

**Comment by lordpidey** [comment]
I think this is a great idea.  Inevitable doom isn't a good idea.  Don't prolong the inevitable, either give a chance to succeed, or have them lose.

**Comment by Kromgar** [comment]
@KorPhaeron 

Stims? You mean the stims that were entirely removed from mining? We ain't got no stims and survival pens have 0 stimulants

**Comment by KorPhaeron** [comment]
@Kromgar 

Line 145 of hypospray.dm, it has nanites (adminrozine)

**Comment by KorPhaeron** [comment]
@Kromgar 

Forgot the nanites got nerfed. It still puts you back to full health/speed fairly quickly though

**Comment by BlueNothing** [comment]
I think this is a great idea, too. The only other jobs that are anywhere near this dependent on ping are jobs whose functions include those of Security Officer, and that's PvP. Considering the relatively high risk (and significant reward, but in particular, the risk) that mining entails, removing slowdowns when fighting bosses seems reasonable.

**Comment by Repukan** [comment]
>I think this is a great idea, too. The only other jobs that are anywhere near this dependent on ping are jobs whose functions include those of Security Officer, and that's PvP. Considering the relatively high risk (and significant reward, but in particular, the risk) that mining entails, removing slowdowns when fighting bosses seems reasonable.

Except miners don't have to fight monsters, at all, to fight. It is very possible to mine a large area and volume of minerals without ever firing a kinetic accelerator offensively.

Comparing miners and security officers is a false equivalence since security job entails tasing people and thus it is affected by ping. Miners don't need to hunt monsters, or fight the big dangerous monsters to mine. They can easily be avoided, and a miner and player can (hopefully) recognize when their internet is shitting itself to not engage those creatures.

As I said before, you don't buff or nerf PVP to compensate for something not caused by game design., because it just makes poorer and shittier design. It would only lead to those that don't lag to be able to abuse the items that are hidden behind these creatures since it is an incredible buff for them, not for the lagging user. If you're lagging and you get hit, the slowdown won't kill you. The stuttering, slideshow-y movement, the teleportation of the monster on top of you due to latency, or running into lava because now you're too fast will.

**Comment by Hatfish** [comment]
It shouldn't cut slowdowns out entirely, maybe give just enough reduction to give you the time to juke the megafauna and medipen up.

**Comment by KorPhaeron** [comment]
Unless you got straight up instant killed by a colossus shotgun I'm not sure how anyone would be able to actually die fighting such slow monsters if they could just leave at full speed whenever they wanted

**Comment by Xhuis** [comment]
The change is meant to allow them to keep fighting at low health, not easily retreat; with stuff like the Hierophant retreat is rendered impossible through it making an arena around you, but for other megafauna I'm not sure. It'd be easy to add some sort of mechanic to stop the adrenaline if you go too long without damaging the fauna or get to a certain distance.

**Comment by Repukan** [comment]
>Hierophant retreat is rendered impossible through it making an arena around you

Uh, that's factually incorrect

**Comment by Xhuis** [comment]
Rendered *mostly* impossible. If you have external methods to escape it, it's possible, but otherwise you're boxed in until the arena dissipates.

You must be fun at parties.

**Comment by ExcessiveUseOfCobblestone** [comment]
Can people please respect lavaland being hardmode? This is really one of the bigger draws to players, including myself.

I know you may not like it but that's a huge draw to it and padding for the sake of "but sometimes I might die" for something entirely optional to begin with seems a bit silly.

**Comment by Xhuis** [comment]
It could very well be that I've played enough miner that it isn't hard for me anymore and this is me unconsciously trying to trivialize it further. I never considered that, but it does seem likely now that it's been pointed out.

Locking this behind mining points *does* seem like a good idea to me, but at that point the survival medipens would be better. I'll have to think on it.

**Comment by ExcessiveUseOfCobblestone** [comment]
perhaps have an object that negates slowdown if you're within screen range (no LoS) that you can setup around the boss. Bonus points if the bosses will try to destroy them to hinder you.

**Comment by Repukan** [comment]
Could make it a syringe that injects you with enough of it for x minutes of combat so you fight against the clock too.

**Comment by KorPhaeron** [comment]
Also you can just bring a friend (or buy one in the mining vendor) and they can drag you clear when you're hurt

**Comment by Xhuis** [comment]
I think the anchored object is probably the best solution - a huge bulky generator that induces Adrenaline in you, and counts as a valid target. Delivered via beacon like syndibombs and the like.

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 1]
```diff
@@ -68,6 +68,18 @@
 				grant_achievement(medal_type,score_type)
 		..()
 
+/mob/living/simple_animal/hostile/megafauna/Life()
+	..()
+	if(vision_range == aggro_vision_range) //This is the only way we can check aggro due to >mobcode
+		var/list/PT = ListTargets()
+		for(var/mob/living/L in PT)
+			if(L.client)
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: Use a human check instead

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by KorPhaeron** [comment]
We have entire departments dedicated to building weapons and chems and other bullshit that never gets used because there are no appropriate targets to use it on, we shouldn't remove a potential use for them because Repukan/Ergo have skewed our idea of who is supposed to kill bosses and with what

**Comment by Xhuis** [comment]
Works for me. Worth a shot!

**Pull request closed without merging**